movement_primitives/dmp/_dmp.py

```diff
@@ -416,8 +422,31 @@ def __init__(self, n_dims, execution_time=1.0, dt=0.01,
 
         self._init_forcing_term()
 
-        self.alpha_y = 25.0
-        self.beta_y = self.alpha_y / 4.0
+        if isinstance(alpha_y, float):
```

Comment:
* This whole thing looks a bit fragile. What if the input is a jax.numpy.ndarray or a tuple? What if the input is an integer (e.g., `25`)? I would like to avoid manual `isinstance` checks and `type` calls.
    * `isinstance(..., float)` should be replaced by checking for a scalar numerical value and cast it to a 1d numpy array of dtype float
    * `isinstance(..., (np.ndarray, list))` should be replaced by checking for an iterable or array-like interface and cast it to a 1d numpy array of dtype float
* I would also suggest to extract this as a more general function that will be applied to `alpha_y` and `beta_y`. 

`sklearn` has a similar function: [`sklearn.utils.check_array`](https://github.com/scikit-learn/scikit-learn/blob/8a2d5ffa0/sklearn/utils/validation.py#L718). Maybe we can use this for inspiration (although it is certainly way to complex for our use case).

Reply:
Please see [9103981](https://github.com/dfki-ric/movement_primitives/pull/59/commits/91039818c8624e4c282220596192a4bb0863932a), and let me know if this works for you?

Reply:
I'd suggest to extract it as a function and give it a more generic name. It's not specific to DMPs.

Reply:
I will change the method name, but any suggestions where to put `_process_gain_input`? Either `movement_primitives/base.py` or I could create `movement_primitives/utils.py` perhaps?